cmd/gcp-controller-manager/node_syncer.go

```diff
@@ -211,29 +215,36 @@ func newNodeMap() *nodeMap {
 	}
 }
 
-func (nm *nodeMap) add(node, pod string, gsa gsaEmail) (gsaEmail, bool) {
+func (nm *nodeMap) add(node string, podUID types.UID, podKey string, gsa gsaEmail) (gsaEmail, bool) {
 	nm.Lock()
 	defer nm.Unlock()
 	n, found := nm.m[node]
 	if !found {
-		nm.m[node] = map[string]gsaEmail{pod: gsa}
+		p := pod{podKey, gsa}
+		nm.m[node] = map[types.UID]pod{podUID: p}
 		return "", false
 	}
-	g, found := n[pod]
-	n[pod] = gsa
-	return g, found
+	p, found := n[podUID]
+	n[podUID] = pod{podKey, gsa}
+	return p.gsa, found
```

Comment:
When found is false, p will nil. Will p.gsa trigger panic?

Reply:
Great catch!